src/data/script.js

```diff
@@ -450,16 +453,22 @@ function checkPost(post) {
 	}
 
 	var postText = '';
-	postText += post.querySelector('.post_header').innerHTML.replace(noTags, ' ');
-	postText += post.querySelector('.post_content').innerHTML;
+	var postHeader = post.querySelector('.post_header');
+	var postContent = post.querySelector('.post_content');
+	var postTags = post.querySelector('.post_tags');
 
-	if (post.querySelector('.post_tags')) {
-		postText += post.querySelector('.post_tags').innerHTML.replace(noTags, ' ');
-	}
+	if (postHeader && !settings.ignore_header)
+		postText += postHeader.innerHTML.replace(noTags, ' ');
+
+	if (postContent && !settings.ignore_body)
+		postText += postContent.innerHTML.replace(noTags, ' ');
```

Comment:
You may not realize it, but you're making a big change to the default functionality of Tumblr Savior here.